lib/langchain/llm/openai.rb

```diff
@@ -60,14 +60,37 @@ def complete(prompt:, **params)
     end
 
     #
-    # Generate a chat completion for a given prompt
+    # Generate a chat completion for a given prompt or messages.
+    #
+    # == Examples
+    #
+    #     openai.chat prompt: "When was Ruby first released?"
+    #
+    #     openai.chat context: "You are RubyGPT, a helpful chat bot for helping people learn Ruby", prompt: "Does Ruby have a REPL like IPython?"
+    #
+    #     openai.chat messages: [
+    #       {
+    #         role: "system",
```

Comment:
You don't actually have to pass a `system` message. You can pass `context` instead and the client will prepend a system message to message history.

Reply:
Right, you don't have to, but this is also a valid way to chat. I can include more examples.

I think I would tend to prefer this way, since it directly maps to the OpenAI REST API: https://platform.openai.com/docs/api-reference/chat/create